### ISSUE TITLE
Sometimes countdown converted into eta imporperly

### ISSUE THREAD

**Issue by lexabug** [celery/celery#3247]
In our app we are using celery 3.1.18.
We are using chords to make a call after all chords subtasks are complete. And chord unlock handler is called every 5 seconds. Yesterday I noticed that some one chord got stuck and found the reason why.
Below is info log of the worker handling chord unlock function:

```
2016-06-09 07:11:50,843 - avaaz_common.queue - INFO - [29462] callback: {u'chord_size': None, u'task': u'avaaz.callback', u'args': [u'all_chunks_sent', {u'blast_process_ids': [536272]}], u'optio
ns': {}, u'subtask_type': None, u'kwargs': {}, u'immutable': False}
2016-06-09 07:11:51,619 - avaaz_common.queue - INFO - [29462] completed_count: 79/288
2016-06-09 07:11:51,629 - celery.worker.job - INFO - [22498] Task avaaz.chord_unlock[b6d2dc41-3c80-45cc-828c-9a276dcc95b0] retry: Retry in 5s
2016-06-09 07:11:51,631 - celery.worker.strategy - INFO - [22498] Received task: avaaz.chord_unlock[b6d2dc41-3c80-45cc-828c-9a276dcc95b0] eta:[2016-06-09 07:11:56.624274]

...

2016-06-09 07:11:56,848 - avaaz_common.queue - INFO - [29462] callback: {u'chord_size': None, u'task': u'avaaz.callback', u'args': [u'all_chunks_sent', {u'blast_process_ids': [536272]}], u'options': {}, u'subtask_type': None, u'kwargs': {}, u'immutable': False}
2016-06-09 07:11:57,666 - avaaz_common.queue - INFO - [29462] completed_count: 81/288
2016-06-09 07:11:57,674 - celery.worker.job - INFO - [22498] Task avaaz.chord_unlock[b6d2dc41-3c80-45cc-828c-9a276dcc95b0] retry: Retry in 5s
2016-06-09 07:11:57,676 - celery.worker.strategy - INFO - [22498] Received task: avaaz.chord_unlock[b6d2dc41-3c80-45cc-828c-9a276dcc95b0] eta:[2016-06-09 11:12:02.671058]
```

Pay attention to lines at `2016-06-09 07:11:51,631`  and `2016-06-09 07:11:57,676`. Both lines are displaying received task with ETA attribute. In both cases the task should have been retried in 5 seconds. But in the second case ETA got +4 hours.
We have `CELERY_ENABLE_UTC: False` and the app is running on the sever with system timezone of 'EDT'. Since the diff between proper ETA and that i saw is 4  hours I assume that ETA was converted into UTC timestamp.
Could you please point me to the place in code where countdown is converted into ETA before publishing retried task onto queue back?

**Comment by ask** [comment]
The countdown is converted here into datetime and isoformat here:
https://github.com/celery/celery/blob/3.1/celery/app/amqp.py#L258-L263

The worker converts the ETA back into datetime here:
https://github.com/celery/celery/blob/3.1/celery/worker/strategy.py#L67-L76

**Comment by auvipy** [comment]
plz verify this after 4.2 release

**Comment by lexabug** [comment]
When 4.2 is going to be released?

**Comment by auvipy** [comment]
see just 1 issue remain that's documentation related https://github.com/celery/celery/milestone/19

**Comment by lexabug** [comment]
Was that fixed in 4.1? My app is built with django 2.0 and celery 4.1. Config has the following lines:
```
TIME_ZONE = 'America/New_York' # django's timezone
...
CELERY_TIMEZONE = 'America/New_York' #celery's timzone
```
My default retry delay is 5s. And when task is retried i see the following line:
```
[2018-01-19 04:22:11,917: INFO/MainProcess] Received task: validate_user_email[php_5a61b8c3b731f7.05616518]  ETA:[2018-01-19 09:22:16.825518-04:56]
```
But afte 5 seconds task is not retried. Seems like ETA of retried task is misinterpreted when non-UTC tz is used.

**Comment by lexabug** [comment]
Well, I solved my problem with using the UTC tz for the celery tasks only. It's not so crucial for the app on the whole.
